src/components/Experiment/QuickInput/Tabs/TextInput/useTextInputControl.js

```diff
@@ -1,25 +1,28 @@
-import {useState} from "react";
+import { useState } from "react";
 import Task from "../../../../../helpers/Task";
 
 export default function useTextInputControl(props) {
     const [text, setText] = useState(props.values[0]);
     const task = Task.getStaticTask(props.task);
 
-    const textChanged = async (event) => {
+    const textChanged = async (event, index = null) => {
         if (event.persist) {
             event.persist();
         }
 
         setText(event.target.value);
 
-        if (typeof(props.inputSelected) === "function") {
-            props.inputSelected(text, 0);
+        if (typeof (props.inputSelected) === "function") {
+            if (index === null) {
+                console.error("Index is required for multi-input tasks");
```

Comment:
Asking the question above just to make sure that this error doesn't get thrown in single-input tasks.

Reply:
My bad, I thought ` if (typeof (props.inputSelected) === "function") ` is a check for multi-input. I have fixed that now.